File: backend/src/support/entities/support-request.entity.ts
export class SupportRequest {
  id: number;
  userId: number;
  subject: string;
  message: string;
  status: 'OPEN' | 'IN_PROGRESS' | 'RESOLVED' | 'CLOSED';
  adminReply?: string;
  createdAt: string;
  updatedAt: string;
}

File: Code/app/src/main/java/com/group_7/library_management/data/local/entity/SupportRequestEntity.kt
package com.group_7.library_management.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "support_requests")
data class SupportRequestEntity(
    @PrimaryKey
    val id: Long,
    val userId: Long,
    val bookId: Long?,
    val bookTitle: String?,
    val subject: String,
    val message: String,
    val status: String,
    val adminReply: String?,
    val repliedAt: String?,
    val createdAt: String,
    val updatedAt: String
)
